internal/yunikorn/event_handler.go

```diff
@@ -234,3 +235,47 @@ func (s *Service) handleAppRemoveEvent(ctx context.Context, ev *si.EventRecord)
 			ev.GetEventChangeDetail())
 	}
 }
+
+func (s *Service) handleQueueEvent(ctx context.Context, ev *si.EventRecord) {
+	logger := log.FromContext(ctx)
+
+	switch ev.GetEventChangeType() {
+	case si.EventRecord_ADD:
+		s.queueAddAccumulator.add(ev)
+	case si.EventRecord_SET:
+		// Ignored for now
+	case si.EventRecord_REMOVE:
+		// Ignored for now
+	case si.EventRecord_NONE:
+	default:
+		// should be warning
+		logger.Warnf("unknown event EventChangeType for an Event of type QUEUE: %v", ev.GetEventChangeType())
+	}
+}
```

Comment:
According to https://github.com/G-Research/yunikorn-history-server/issues/186, the sync logic will manage changes in the queues, including Add, Delete, and Update operations. 

Can we say that the same accumulator can be used for all the above cases?

Reply:
Yes, but each of these cases would require special logic. What I wanted to do is prepare the event store for the Add event, and then handle deletion events in a separate PR. This PR is pretty big right now, so I would ideally want to break it down.
Breaking it down would make it easier to review and explain changes.